exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/InternalTraceExporter.java

```diff
@@ -0,0 +1,174 @@
+
+      // We only use the batchWriteSpans API in this exporter.
+      builder
+          .batchWriteSpansSettings()
+          .setSimpleTimeoutNoRetries(
+              org.threeten.bp.Duration.ofMillis(configuration.getDeadline().toMillis()));
+      // For testing, we need to hack around our gRPC config.
+      if (configuration.getInsecureEndpoint()) {
+        builder.setCredentialsProvider(NoCredentialsProvider.create());
+        builder.setTransportChannelProvider(
+            FixedTransportChannelProvider.create(
+                GrpcTransportChannel.create(
+                    ManagedChannelBuilder.forTarget(configuration.getTraceServiceEndpoint())
+                        .usePlaintext()
+                        .build())));
+      } else {
+        Credentials credentials =
+            configuration.getCredentials() == null
+                ? GoogleCredentials.getApplicationDefault()
+                : configuration.getCredentials();
+        builder.setCredentialsProvider(
+            FixedCredentialsProvider.create(checkNotNull(credentials, "credentials")));
+        builder.setEndpoint(configuration.getTraceServiceEndpoint());
+        builder.setHeaderProvider(HEADER_PROVIDER);
+      }
+
+      return new InternalTraceExporter(
+          projectId,
+          new CloudTraceClientImpl(TraceServiceClient.create(builder.build())),
+          configuration.getAttributeMapping(),
+          configuration.getFixedAttributes());
+    }
+    return InternalTraceExporter.createWithClient(
+        projectId,
+        new CloudTraceClientImpl(TraceServiceClient.create(stub)),
+        configuration.getAttributeMapping(),
+        configuration.getFixedAttributes());
+  }
+
+  /**
+   * A noop implementation of {@link TraceExporter}. Should be used when {@link TraceExporter}
+   * cannot be initialized due to some error/exception.
+   *
+   * @return noop implementation of {@link TraceExporter} as a {@link SpanExporter}.
+   */
+  static SpanExporter noop() {
+    return new SpanExporter() {
+      @Override
+      public CompletableResultCode export(@Nonnull Collection<SpanData> spans) {
+        return CompletableResultCode.ofSuccess();
+      }
+
+      @Override
+      public CompletableResultCode flush() {
+        return CompletableResultCode.ofSuccess();
+      }
+
+      @Override
+      public CompletableResultCode shutdown() {
+        return CompletableResultCode.ofSuccess();
+      }
+
+      @Override
+      public String toString() {
+        return "NoopTraceExporter{}";
+      }
+    };
+  }
+
+  InternalTraceExporter(
+      String projectId,
+      CloudTraceClient cloudTraceClient,
+      ImmutableMap<String, String> attributeMappings,
+      Map<String, AttributeValue> fixedAttributes) {
+    this.projectId = projectId;
+    this.cloudTraceClient = cloudTraceClient;
+    this.projectName = ProjectName.of(projectId);
+    this.translator = new TraceTranslator(attributeMappings, fixedAttributes);
```

Comment:
That does sound better, I will open a separate issue that would look into in the existing TODOs of this class and do the necessary refactoring, this would help in keeping the scope of this PR small.